### PR TITLE
Add conversions Symbol <-> ScVal

### PR DESCRIPTION
**Pull request by jonjove**
### What

Added TryFrom Symbol/ScVal.

### Why

Make https://github.com/stellar/rs-stellar-contract-sdk/pull/316 work.

### Known limitations

This is eating conversion errors, in the interest of getting something working as quickly as possible.

### CODE REVIEW

**Comment by jonjove** [comment]
@leighmcculloch @jayz22 I'm going to merge this as soon as possible so I can finish the macros work. We can clean up anything we need regarding errors in a separate PR.

File: stellar-contract-env-common/src/symbol.rs [hunk 1]
```diff
@@ -282,6 +282,46 @@ impl FromIterator<char> for Symbol {
     }
 }
 
+use crate::xdr::ScVal;
+
+#[cfg(feature = "std")]
+impl TryFrom<ScVal> for Symbol {
+    type Error = ();
```
Comment (leighmcculloch) [inline stellar-contract-env-common/src/symbol.rs:289]: ConversionError?

File: stellar-contract-env-common/src/symbol.rs [hunk 2]
```diff
@@ -282,6 +282,46 @@ impl FromIterator<char> for Symbol {
...
+        ))
+    }
+}
+
+#[cfg(feature = "std")]
+impl TryFrom<&Symbol> for ScVal {
+    type Error = ();
+    fn try_from(s: &Symbol) -> Result<Self, Self::Error> {
+        s.clone().try_into()
+    }
```
Comment (leighmcculloch) [inline stellar-contract-env-common/src/symbol.rs:322]: I usually try to do these in reverse, implement for the ref version, and then have the other one cast to it. You're collecting in the non-ref version anyway, so then the ref version becomes 2x clones.

File: stellar-contract-env-common/src/symbol.rs (hunk 1 above)
Comment (jonjove) [reply stellar-contract-env-common/src/symbol.rs:289]: We'll clean it up in follow-up work. I'm not up to date on the best practices regarding error propagation in the host crate.

**Review by leighmcculloch — APPROVED** [review]
Could you fix the error to be ConversionError? Otherwise LGTM.

I note an optimization, but nbd.

File: stellar-contract-env-common/src/symbol.rs (hunk 2 above)
Comment (jonjove) [reply stellar-contract-env-common/src/symbol.rs:322]: Does the reverse work in this case? I'm calling `into_iter` which takes ownership. Is there a non-into option here?

File: stellar-contract-env-common/src/symbol.rs (hunk 1 above)
Comment (jonjove) [reply stellar-contract-env-common/src/symbol.rs:290]: Opened #260

**Pull request merged**